src/averageAges.js

```diff
@@ -55,10 +77,36 @@ function calculateWomenAverageAge(people, withChildren) {
  * @return {number}
  */
 function calculateAverageAgeDiff(people, onlyWithSon) {
-  // 1. find a mother of each person (or only for men)
-  // 2. keep people who have mothers in the array
-  // 3. calculate the difference child.born - mother.born
-  // 4. return the average value
+  const mothers = people.reduce((acc, person) => {
+    if (person.mother !== null && !acc.includes(person.mother)) {
+      acc.push(person.mother);
+    }
+
+    return acc;
+  }, []);
+
+  const mothersWithSons = people.filter(person =>
+    mothers.includes(person.name));
```

Comment:
This could be simplified, to get all mothers you can write:
```js
const mothers = people.filter(mother => 
  people.some(child => mother.name === child.mother)
)
```